index.d.ts

```diff
@@ -0,0 +1,56 @@
+import { ActionResult, ActionType, ActionsType, VNode, View } from 'hyperapp'
+
+export as namespace hyperapp__fx
+
+// hyperapp
+
+export interface App {
+    <State, Actions>(
+        state: State,
+        actions: ActionsType<State, Actions>,
+        view: View<State, Actions>,
+        container: Element | null,
+    ): Actions
+}
+
+// @hyperapp/fx
+
+export type EffectTuple = [string, object]
+
+export type Effect = EffectTuple | Array<EffectTuple>
```

Comment:
Is there a way to write the type to show that `EffectTuple`s may be deeply nested and not just a single dimension array?

Reply:
It’s possible to make the type recursive, yes. I guess I was taking a shortcut.